### PR TITLE
INT-11 [BREAKING CHANGES] Migrate to ESM, update supported Node.js to v18

### PR DESCRIPTION
**Pull request by batovpasha**

### CODE REVIEW

**Comment by batovpasha** [comment]
@vladholubiev Could you please setup the CircleCI project for this repo?
image

File: package.json [hunk 1]
```diff
@@ -54,8 +51,11 @@
     "fast-lorem-ipsum": "1.2.0",
     "husky": "9.0.11",
     "jest": "29.7.0",
+    "jest-junit": "16.0.0",
     "lint-staged": "15.2.2",
     "prettier": "3.2.5",
+    "ts-jest-resolver": "2.0.1",
+    "tsx": "4.7.1",
```
Comment (vladholubiev) [inline package.json:58]: why do we need `tsx` here?

**Comment by vladholubiev** [comment]
> @vladholubiev Could you please set up the CircleCI project for this repo? 

done

File: package.json (hunk 1 above)
Comment (batovpasha) [reply package.json:58]: image

**Review by vladholubiev — APPROVED** [review]

**Pull request merged**